src/coreclr/gc/gc.cpp

```diff
@@ -49307,3 +49377,294 @@ void PopulateDacVars(GcDacVars *gcDacVars)
 #endif // MULTIPLE_HEAPS
     gcDacVars->generation_field_offsets = reinterpret_cast<int**>(&generation_field_offsets);
 }
+
+void GCHeap::RefreshMemoryLimit()
+{
+    gc_heap::refresh_memory_limit();
+}
+
+void gc_heap::configure_memory_settings(bool is_initialization, uint32_t& nhp, uint32_t& nhp_from_config, size_t& seg_size_from_config)
+{
+    // If the hard limit is specified, the user is saying even if the process is already
+    // running in a container, use this limit for the GC heap.
+    if (!hard_limit_config_p)
+    {
+        if (is_restricted_physical_mem)
+        {
+            uint64_t physical_mem_for_gc = total_physical_mem * (uint64_t)75 / (uint64_t)100;
+#ifndef USE_REGIONS
+            if (is_initialization)
```

Comment:
would you mind adding a comment here to explain the reason why we do this for segments?